### ISSUE TITLE
AttributeError when uploading image with python3

### ISSUE THREAD

**Issue by eduard-sukharev** [adi-/django-markdownx#6]
Installed markdownx version is 1.2. Python version is 3.4.0, project is in virtualenv.

drag'n'drop image from local folder shows 500 error on request to server in developer console:

```
TypeError at /markdownx/upload/
string argument expected, got 'bytes'

Request Method: POST
Request URL: http://localhost:8000/markdownx/upload/
Django Version: 1.8.5
Python Executable: /home/kraplax/noslack/bin/python3
Python Version: 3.4.0
Python Path: ['/home/kraplax/noslack/src', '/home/kraplax/noslack/lib/python3.4', '/home/kraplax/noslack/lib/python3.4/plat-i386-linux-gnu', '/home/kraplax/noslack/lib/python3.4/lib-dynload', '/usr/lib/python3.4', '/usr/lib/python3.4/plat-i386-linux-gnu', '/home/kraplax/noslack/lib/python3.4/site-packages']
Server time: Sat, 17 Oct 2015 22:59:28 +0000
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'time_sections',
 'account',
 'markdownx',
 'ublog',
 'pinax_theme_bootstrap',
 'bootstrapform')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'account.middleware.LocaleMiddleware',
 'account.middleware.TimezoneMiddleware')

Traceback:
File "/home/kraplax/noslack/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  132.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/home/kraplax/noslack/lib/python3.4/site-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)
File "/home/kraplax/noslack/lib/python3.4/site-packages/django/views/generic/base.py" in dispatch
  89.         return handler(request, *args, **kwargs)
File "/home/kraplax/noslack/lib/python3.4/site-packages/django/views/generic/edit.py" in post
  215.             return self.form_valid(form)
File "/home/kraplax/noslack/lib/python3.4/site-packages/markdownx/views.py" in form_valid
  30.         image_path = form.save()
File "/home/kraplax/noslack/lib/python3.4/site-packages/markdownx/forms.py" in save
  28.         img.save(thumb_io,  self.files['image'].content_type.split('/')[-1].upper())
File "/home/kraplax/noslack/lib/python3.4/site-packages/PIL/Image.py" in save
  1665.             save_handler(self, fp, filename)
File "/home/kraplax/noslack/lib/python3.4/site-packages/PIL/PngImagePlugin.py" in _save
  690.     fp.write(_MAGIC)

Exception Type: TypeError at /markdownx/upload/
Exception Value: string argument expected, got 'bytes'
Request information:
GET: No GET data
```

**Comment by adi-** [comment]
Could you confirm, that fix in branch `Issue#6` fixed the issue?

**Comment by eduard-sukharev** [comment]
i've tried fixing it that way, but it then chokes at line 31 and says that `thumb_io` has no member `len`

**Comment by eduard-sukharev** [comment]
the solution from [here](http://stackoverflow.com/a/4677542/1808052) solves the missing `len` attribute: as explained [further in comments](http://stackoverflow.com/questions/4677433/in-python-how-do-i-check-the-size-of-a-stringio-object#comment20721136_4677448) to another solution:

Also note that the len was removed in Python3, so you cannot use your solution in Python3.

**Comment by adi-** [comment]
Fixed in v1.2.1
